### PR TITLE
[SPARK-14441] [SQL] Consolidate DDL tests

### PR DESCRIPTION
**Pull request by bomeng**
## What changes were proposed in this pull request?

Today we have `DDLSuite`, `DDLCommandSuite`, `HiveDDLCommandSuite` and `HiveDDLSuite`. In this PR, I am trying to consolidate the files as much as possible. Two files are left for now, since it is good to put Hive related test suite in the Hive package. 

Along with the combination, I also did some modification of current codes mainly in order to improve the codings, here is the summary:
1. Use `.contains(...)` instead `== Some(...)` for Options, this method is introduced in Scala 2.11 and it is recommended method to use for this purpose;
2. Make map consistent to use `->`, instead of 2 elements tuple;
3. Use `isEmpty()` instead of `== None`
4. Add `private` to the parser and narrow the scope of implicit (put inside one of the tests); 
5. Modify the names of some tests to be unique, since they are in one file now;
## How was this patch tested?

I did not change the logic of any tests, just move around and improve the codes, so existing test cases should remain same.

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveDDLCommandSuite.scala [hunk 1]
```diff
@@ -113,10 +244,10 @@ class HiveDDLCommandSuite extends PlanTest {
 
     val (desc, exists) = extractTableDesc(s2)
     assert(exists)
-    assert(desc.identifier.database == Some("mydb"))
+    assert(desc.identifier.database.contains("mydb"))
```
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveDDLCommandSuite.scala]: All these similar changes will break the Scala 2.10 build

**Comment by gatorsmile** [comment]
```Use .contains(...) instead == Some(...) for Options, this method is introduced in Scala 2.11 and it is recommended method to use for this purpose;```
This should not be used in Spark code. It will break the 2.10 build. 

See my PR: https://github.com/apache/spark/pull/12201

**Comment by gatorsmile** [comment]
To be honest, I do not know why we need to merge these test case files. Their purposes are different. One is to verify the functionalities of parsers; another is to verify the execution of DDL/commands. In the future, we might add more test cases to both files. The files will grow bigger and bigger.  

cc @yhuai @andrewor14

**Comment by andrewor14** [comment]
@gatorsmile that's a good point. I think a better goal of the issue is to make the tests consistent and make sure the functionality of the test suites are clearly isolated. I don't think just literally merging all the test files together is a great idea.

@bomeng also we should wait for all the other DDLs tasks have finished before we start working on this one. Otherwise there will be a lot of conflicts and the later PRs will still put them in the wrong places.

**Comment by bomeng** [comment]
Ok, not a problem. Thanks.

**Comment by rxin** [comment]
Yea the current tests are pretty scattered and they are way too big to merge into a single one. Can you just close this pull request? Thanks.

**Comment by bomeng** [comment]
closing this PR. thanks.

**Pull request closed without merging**